File: platform/macos/monarch_macos/MonarchState.swift
//
//  MonarchState.swift
//  monarch_macos
//
//  Created by Fernando Trigoso on 5/9/22.
//

import Foundation

struct MonarchState {
    let device: DeviceDefinition
    let scale: ScaleDefinition
    let dock: DockSide

    
    init(device: DeviceDefinition, scale: ScaleDefinition, dock: DockSide) {
        self.device = device
        self.scale = scale
        self.dock = dock
    }
}

extension MonarchState : InboundChannelArgument {
    
    init(standardMap args: [String : Any]) {
        self.init(
            device: DeviceDefinition.init(standardMap: args["device"] as! [String : Any]),
            scale: ScaleDefinition.init(standardMap: args["scale"] as! [String: Any]),
            dock: dockFromString(dockString: args["dock"] as! String)
        )
    }
}

File: platform/windows/src/monarch_state.cpp
#include <vector>
#include "monarch_state.h"
#include "channels.h"

MonarchState::MonarchState(EncodableMap args)
{
	device = DeviceDefinition{ std::get<EncodableMap>(MapUtils::getValue(args, "device")) };
	scale = StoryScaleDefinition{ std::get<EncodableMap>(MapUtils::getValue(args, "scale")) };
	dock = dockFromString(std::get<std::string>(MapUtils::getValue(args, "dock")));
}
